lang/llvm-12/Portfile

```diff
@@ -0,0 +1,424 @@
+worksrcdir              ${worksrcdir}/llvm
+
+# hand-tweaked, approximately c++ standard 2014
+compiler.blacklist      *gcc* {clang < 801} macports-clang-3.*
+compiler.fallback       clang macports-clang-9.0
+
+# We can't use Ninja with our custom destrooting plan for llvm
+#cmake.generator         Ninja
+cmake.install_prefix    ${sub_prefix}
+cmake.build_type        Release
+
+# clean up some cmake-PG added values we don't want to use, because we want @rpaths
+cmake.install_rpath
+
+configure.pre_args-delete \
+    -DCMAKE_INSTALL_NAME_DIR=${cmake.install_prefix}/lib \
+    -DCMAKE_INSTALL_RPATH=${cmake.install_prefix}/lib
+
+configure.pre_args-replace \
+    -DCMAKE_BUILD_WITH_INSTALL_RPATH:BOOL=ON \
+    -DCMAKE_BUILD_WITH_INSTALL_RPATH:BOOL=OFF
+
+configure.pre_args-replace \
+    -DCMAKE_SYSTEM_PREFIX_PATH="${prefix}\;/usr" \
+    -DCMAKE_SYSTEM_PREFIX_PATH="${cmake.install_prefix}\;${prefix}\;/usr"
+
+configure.args-append \
+    -DLLVM_LINK_LLVM_DYLIB=ON \
+    -DLLVM_ENABLE_RTTI=ON \
+    -DLLVM_INCLUDE_TESTS=OFF \
+    -DLLVM_INCLUDE_EXAMPLES=OFF \
+    -DLLVM_ENABLE_FFI=ON \
+    -DFFI_INCLUDE_DIR=${prefix}/include \
+    -DFFI_LIBRARY_DIR=${prefix}/lib
+
+# Set CMAKE_LIBTOOL if we're using MacPorts-provided cctools
+if {[lsearch -exact $PortInfo(depends_build) port:cctools] != -1} {
+    configure.args-append \
+        -DCMAKE_LIBTOOL=${prefix}/bin/libtool
+}
+
+depends_build-append    port:python39
+set pythonfullpath      ${prefix}/bin/python3.9
+configure.args-append   -DPython3_EXECUTABLE=${pythonfullpath}
+
+patch.dir ${workpath}/${distname}
+patch.pre_args  -p1
+
+patchfiles-append \
+    0001-MacPorts-only-clang-use-MP-omp-locations.patch \
+    0002-MacPorts-Only-Don-t-embed-the-deployment-target-in-t.patch \
+    0003-MacPorts-Only-Prepare-clang-format-for-replacement-w.patch \
+    0004-MacPorts-Only-Fix-name-of-scan-view-executable-insid.patch \
+    0006-Define-EXC_MASK_CRASH-and-MACH_EXCEPTION_CODES-if-th.patch \
+    0007-Threading-Only-call-pthread_setname_np-if-we-have-it.patch \
+    0008-Threading-Only-call-setpriority-PRIO_DARWIN_THREAD-0.patch \
+    0009-lib-Support-Unix-Path.inc-define-COPYFILE_CLONE-if-n.patch \
+    0010-compiler-rt-cmake-config-ix.cmake-was-Leopard-No-ASA.patch \
+    0011-Fix-missing-long-long-math-prototypes-when-using-the.patch \
+    0012-compiler-rt-add-some-defs-missing-in-older-SDKs.patch \
+    0013-clang-add-back-runtime-libraries-used-on-10.4-and-10.patch \
+    0014-Fix-float.h-to-work-on-Snow-Leopard-and-earlier.patch \
+    0015-Fixup-libstdc-header-search-paths-for-older-versions.patch
+
+if {${os.platform} eq "darwin" && ${os.major} < 14} {
+    patchfiles-append \
+        0016-10.10-and-less-compiler-rt-find-macosx-sdk.patch \
+}
+
+if {${os.platform} eq "darwin" && ${os.major} < 11} {
+    patchfiles-append \
+        0017-10.6-and-less-libcxx-work-around-no-reexported_symbo.patch \
+        0018-10.6-and-less-clang-default-to-libc-on-all-darwin-sy.patch \
```

Comment:
whoopsy -- this one should be at least 10.8 and less, as we need to catch the 10.7 and 10.8 systems too.